keras/backend/mxnet_backend.py

```diff
@@ -1849,7 +1857,12 @@ def greater(x, y):
         scalar = True
     if scalar:
         return KerasSymbol(mx.sym.Cast(x > y, dtype='uint8'))
-    return KerasSymbol(mx.sym.Cast(mx.sym.broadcast_greater(lhs=x, rhs=y), dtype='uint8'))
+    print("[Debug]", x)
+    print("[Debug]", y)
+    if isinstance(x, mx.sym.Symbol) and isinstance(y, mx.sym.Symbol):
+        return KerasSymbol(mx.sym.Cast(mx.sym.broadcast_greater(lhs=x, rhs=y), dtype='uint8'))
+    else:
+        raise TypeError
```

Comment:
Please add a good message for these type error

Reply:
added.